test/dataloader2/test_dataloader2.py

```diff
@@ -235,6 +235,7 @@ def _collect_data(self, datapipe, reading_service_gen):
             result.append(row)
         for row in dl:
             result.append(row)
+        dl.shutdown()
```

Comment:
This doesn't fail if this is removed, but I think we should add it anyway.

Reply:
I am going to send a fix to it as it seems also breaking the tests for sequentialRS

Reply:
Makes sense to me. I don't think it is breaking this PR though? Or did I miss some CI failure